app/actions/requests/BasicAuthenticatedRequest.scala

```diff
@@ -36,11 +36,25 @@ case class BasicAuthenticatedRequestWithAffinityGroup[A](
       affinityGroup: AffinityGroup,
       request: Request[A]
 ) extends AuthenticatedRequest[A](request)
+
 case class BasicAuthenticatedRequest[A](
       organisationAccount: GroupAccount,
       individualAccount: DetailedIndividualAccount,
       request: Request[A]
-) extends AuthenticatedRequest[A](request)
+) extends AuthenticatedRequest[A](request) with CcaWrappedRequest {
+  override def optAccounts: Option[Accounts] =
+    Some(Accounts(organisationAccount, individualAccount))
+
+  override def isLoggedIn: Boolean = optAccounts.nonEmpty
+
+  override def yourDetailsName: Option[String] = optAccounts.map { acc =>
+    if (s"${acc.person.details.firstName} ${acc.person.details.lastName}" == acc.organisation.companyName) {
+      s"${acc.person.details.firstName} ${acc.person.details.lastName}"
+    } else {
+      s"${acc.person.details.firstName} ${acc.person.details.lastName} - ${acc.organisation.companyName}"
+    }
+  }
```

Comment:
This code should be in CaWrappedRequest as it is in vmv-frontend. Did you remove from `AppointAgentSessionRequest` ?

Reply:
no I didn't remove AppointAgentSessionRequest. AppointAgentSessionRequest carries the session data required for the add agent journey. The manage agent pages don't use AppointAgentSessionRequest, they don't have any session data saved between pages. It only needs the authenticated user's details

Reply:
I think you misunderstood. The implementation of `yourDetailsName` is in the parent trait CCaWrappedRequest in vmv-frontend (where you took it from originally). Previously you had an identical implementation of `yourDetailsName` in AppointAgentSessionRequest.

I'm asking if you have removed that implementation from AppointAgentSessionRequest (otherwise it's duplicated)? And I'm suggesting moving this implementation to CCaWrappedRequest - as it is in vmv-frontend

Reply:
yep, I definitely misunderstood you. I agree with your idea